src/librustc_mir/borrow_check/mod.rs

```diff
@@ -1639,10 +1639,18 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
                                             } else {
                                                 self.get_default_err_msg(place)
                                             };
+                                            let sp = self.mir.source_info(locations[0]).span;
+                                            let mut help_stmt = String::new();
+                                            if let Ok(src) =
+                                                self.tcx.sess.codemap().span_to_snippet(sp) {
+                                                    help_stmt = format!(
+                                                        "consider changing this to be a \
+                                                        mutable reference: `&mut {}`", &src[1..]);
```

Comment:
The help message doesn't need to incorporate the suggested code, this is passed as the third argument to `span_suggestion` and the diagnostic machinery deals with this as appropriate.